### ISSUE TITLE
Service Not Running when NetData is Started

### ISSUE THREAD

**Issue by NMcCloud** [firehol/netdata#1918]
I have several services which take time to start, Tomcat, MySql for example; however, when netdata starts, it immediately determines that these services are disabled because of connection refusals.  Is there a way to force netdata to keep trying indefinitely, otherwise, after the server is restarted I must force a restart of netdata five minutes after it comes back up.  Thank You

**Comment by ktsaou** [comment]
hm... sorry, there is no option for this yet.
The plugins need a successful data collection when they start, and they will never retire, but there is no option to bypass that initial check.

If you use systemd, you can edit `/etc/systemd/system/netdata.service` and add the services you need to start before netdata to [this line](https://github.com/firehol/netdata/blob/6f6b9975a1a24661c524f722ec947eb93dd43831/system/netdata.service.in#L3).

If you don't use systemd, you will need to find how to do this with your init system. I think all of them support this feature.

**Comment by NMcCloud** [comment]
Thanks.  I can figure something out.  Ensuring that netdata starts after other dependencies does not guarantee first-time success as said dependencies do not always finish starting before the startup script finishes (e.g. Tomcat simply starts java and terminates the script, taking several seconds before opening the port)

**Comment by l2isbad** [comment]
> force netdata to keep trying indefinitely

```
[~/git/netdata/conf.d/python.d]grep '^[a-z0-9_]*:' * --count | cut -f2 -d:| awk '{tot+=$1} END {print tot}'
95
```

**Comment by ktsaou** [comment]
@l2isbad I am lost. What is 95?

**Comment by l2isbad** [comment]
95 is the number of python jobs at the moment. And the number is constantly increasing. I don't think it's a good idea to force them "keep trying indefinitely". I believe a simple script in the rc.local can do the job 😄

**Comment by ktsaou** [comment]
You are right, of course.

I have thought this problem once in the past. Then I decided that the best solution in this direction would be for the plugins to save on disk something that will let them know they managed in the past to collect data for a job, and load this information back when they start.

This will introduce another problem, since they will now never retire trying to collect data from a source, even if the user has removed a service.

I decided then that the whole idea is just too complicated to be supported and probably will never be capable of covering all corner cases properly, so I totally agree @l2isbad that restarting netdata after all applications have been started is the best solution so far...
